### PR TITLE
Prepare epics for US support frontend switch-over

### PR DESCRIPTION
**Pull request by JustinPinner**
## What does this change?
Adds a new switch to control whether we route US traffic to the new US landing page on support.theguardian.com.

Sets up epics to respond appropriately to the switch setting and the user's location, setting the UI copy, buttons and/or links as required.

## What is the value of this and can you measure success?
We're getting set up to switch the US over to support.theguardian.com. This is the first step towards redefining the acquisition channels.

## Does this affect other platforms - Amp, Apps, etc?
No, web only.

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?
N/A

## Screenshots
**Switch ON, US user, regular epic**
screen shot 2017-10-17 at 16 11 11 - us switch on us user

**Switch OFF, US user, regular epic**
screen shot 2017-10-17 at 16 08 32 - us switch off - us user 1
screen shot 2017-10-17 at 16 08 46 - us switch off - us user 2

**Switch ON, US user, liveblog epic**
screen shot 2017-10-17 at 12 47 50-us switch on - us user

**Switch OFF, US user, liveblog epic**
screen shot 2017-10-17 at 15 59 33 - us switch off - us user 1
screen shot 2017-10-17 at 15 59 44 - us switch off - us user 2

**And for the sake of completeness;**
**UK user, liveblog epic**
screen shot 2017-10-17 at 14 18 35 - uk on - uk user

**AU user, epic liveblog**
screen shot 2017-10-17 at 15 55 48 - us and uk on - au user 1
screen shot 2017-10-17 at 15 55 58 - us and uk on - au user 2

## Tested in CODE?
**YES**

### CODE REVIEW

File: common/app/conf/switches/FeatureSwitches.scala [hunk 1]
```diff
@@ -476,11 +476,21 @@ trait FeatureSwitches {
...
 
   // Owner: Simple & Coherent
-  val SupportFrontendActive = Switch(
+  val UkSupportFrontendActive = Switch(
     SwitchGroup.Feature,
     "uk-supporter-traffic-to-new-support-frontend",
     "When ON, all UK membership/contribute/support links send traffic to support.theguardian.com",
     owners = Seq(Owner.withGithub("justinpinner")),
+    safeState = On,
+    sellByDate = never,
```
Comment (davidfurey) [inline common/app/conf/switches/FeatureSwitches.scala]: These could/should have a sell by date - even if we make it 6 months from now they aren't intended to be permanent.

File: static/src/javascripts/projects/common/modules/commercial/acquisitions-copy.js [hunk 2]
```diff
@@ -39,10 +39,8 @@ export const usLocalisedFlag = {
 export const liveblog = (
     membershipUrl: string,
     contributionsUrl: string,
-    useSupportDomain: boolean = false
+    supportP2: string
```
Comment (davidfurey) [inline static/src/javascripts/projects/common/modules/commercial/acquisitions-copy.js]: Is this "support paragraph 2"?  It is probably obvious to most people as is, but I read it as a P(riority)2 support incident.

**Review by davidfurey — COMMENTED** [review]
Looks good - so much to review because of the flow types!

File: static/src/javascripts/projects/common/modules/commercial/acquisitions-copy.js (hunk 2 above)
Comment (JustinPinner) [reply static/src/javascripts/projects/common/modules/commercial/acquisitions-copy.js]: Ha, yes. It's really just a component of the ask paragraph, the bit with the CTA links. The convention before was to use something called `controlP2` so I borrowed from that. Can change it to something like `ctaLinkSentence` or something, if it's upsetting as-is.

File: common/app/conf/switches/FeatureSwitches.scala (hunk 1 above)
Comment (JustinPinner) [reply common/app/conf/switches/FeatureSwitches.scala]: Fair point. I didn't want there to be any chance of it shutting off without our say so though. I guess a long expiry would do, but we'll be pulling these out again in the next few weeks I hope.

**Comment by svillafe** [comment]
I tested in CODE and the engagement banner, epic and epic liveblog seem that are working.

File: common/app/conf/switches/FeatureSwitches.scala [hunk 3]
```diff
@@ -476,13 +476,23 @@ trait FeatureSwitches {
...
 
   // Owner: Simple & Coherent
-  val SupportFrontendActive = Switch(
+  val UkSupportFrontendActive = Switch(
     SwitchGroup.Feature,
     "uk-supporter-traffic-to-new-support-frontend",
     "When ON, all UK membership/contribute/support links send traffic to support.theguardian.com",
     owners = Seq(Owner.withGithub("justinpinner")),
+    safeState = On,
+    sellByDate = new LocalDate(2018, 10, 17),
```
Comment (rupertbates) [inline common/app/conf/switches/FeatureSwitches.scala]: This sell by date is yesterday, is that correct?

Comment (JustinPinner) [reply common/app/conf/switches/FeatureSwitches.scala]: 2018

Comment (rupertbates) [reply common/app/conf/switches/FeatureSwitches.scala]: Doh! :-)

File: static/src/javascripts/projects/common/modules/commercial/support-utilities.js [hunk 4]
```diff
@@ -3,14 +3,43 @@
...
+const supportButtonCaption = 'Support the Guardian';
+
+const useSupportDomain = (): boolean => liveInUk || liveInUs;
 const selectBaseUrl = (defaultUrl: string = supportBaseURL): string =>
     useSupportDomain() ? supportBaseURL : defaultUrl;
 const selectEngagementBannerButtonCaption = (defaultCaption: string) =>
     useSupportDomain() ? supportButtonCaption : defaultCaption;
 
-export { useSupportDomain, selectBaseUrl, selectEngagementBannerButtonCaption };
+const liveBlogEpicSupportCtaLinkPara = (
```
Comment (joelochlann) [inline static/src/javascripts/projects/common/modules/commercial/support-utilities.js]: I'd be more inclined to export `liveInUk` and `liveInUs` and then have this logic in `acquisitions-copy.js`. Then all the copy stays in one place. Also, you wouldn't have the redundant `membershipUrl` and `contributionsUrl` params on the `liveblog` function

File: common/app/conf/switches/FeatureSwitches.scala (hunk 3 above)
Comment (joelochlann) [reply common/app/conf/switches/FeatureSwitches.scala]: 2018

Comment (joelochlann) [reply common/app/conf/switches/FeatureSwitches.scala]: whoops, late to the party

File: static/src/javascripts/projects/common/modules/commercial/support-utilities.js (hunk 4 above)
Comment (JustinPinner) [reply static/src/javascripts/projects/common/modules/commercial/support-utilities.js]: Ok, testing this now...

**Comment by JustinPinner** [comment]
Am having problems with this locally now. Will re-deploy to CODE to see if it's an environmental thing. There are no errors captured in the browser that are relevant it seems. @joelochlann would you have a look at the last commit diff and see if you spot anything I've done that breaks ALL epics? Ta. It started playing up after merging from `master`

**Comment by svillafe** [comment]
I tested in CODE again after the conflicts were solved and it is working 👍

**Review by joelochlann — APPROVED** [review]
Looks good!

**Pull request merged**